virtme/guest/virtme-init

```diff
@@ -236,42 +238,42 @@ real_sudoers=/etc/sudoers
 if [ ! -e ${real_sudoers} ]; then
     touch ${real_sudoers}
 fi
-tmpfile="`mktemp --tmpdir=/run/tmp`"
-echo "Defaults secure_path=\"/usr/sbin:/usr/bin:/sbin:/bin\"" > $tmpfile
-echo "root ALL = (ALL) NOPASSWD: ALL" >> $tmpfile
+tmpfile="$(mktemp --tmpdir=/run/tmp)"
+echo "Defaults secure_path=\"/usr/sbin:/usr/bin:/sbin:/bin\"" > "$tmpfile"
+echo "root ALL = (ALL) NOPASSWD: ALL" >> "$tmpfile"
 if [[ -n "${virtme_user}" ]]; then
-    echo "${virtme_user} ALL = (ALL) NOPASSWD: ALL" >> $tmpfile
+    printf -- '%s\n' "${virtme_user} ALL = (ALL) NOPASSWD: ALL" >> "$tmpfile"
 fi
 chmod 440 "$tmpfile"
 if [ ! -f "$real_sudoers" ]; then
 touch "$real_sudoers"
 fi
 mount --bind "$tmpfile" "$real_sudoers"
 
-if cat /proc/cmdline |grep -q -E '(^| )virtme.dhcp($| )'; then
+if grep -q -E '(^| )virtme.dhcp($| )' /proc/cmdline; then
     # Make sure all GIDs are allowed to create raw ICMP sockets (this
     # allows to run ping as regular user).
     echo "0 2147483647" > /proc/sys/net/ipv4/ping_group_range
 
     # udev is liable to rename the interface out from under us.
     for d in /sys/bus/virtio/drivers/virtio_net/virtio*/net/*; do
         virtme_net=$(basename "${d}")
-        busybox udhcpc -i "$virtme_net" -n -q -f -s "$(dirname $0)/virtme-udhcpc-script" &
+        busybox udhcpc -i "$virtme_net" -n -q -f -s "$(dirname "$0")/virtme-udhcpc-script" &
```

Comment:
Below you use `dirname -- "$0"`, but not here.

Also, I don't think `$0` will start with `-`, so maybe `--` is not needed. But best to be consistent.

Reply:
Will change it.